cst438_gradebook/src/test/java/com/cst438/JunitTestGradebook.java

```diff
@@ -73,6 +75,172 @@ public class JunitTestGradebook {
 
 	@Autowired
 	private MockMvc mvc;
+	
+	@Test
+	public void addAssignment() throws Exception {
+		
+		MockHttpServletResponse response;
+
+		// mock database data
+		Course course = new Course();
+		course.setCourse_id(TEST_COURSE_ID);
+		course.setSemester(TEST_SEMESTER);
+		course.setYear(TEST_YEAR);
+		course.setInstructor(TEST_INSTRUCTOR_EMAIL);
+		course.setEnrollments(new java.util.ArrayList<Enrollment>());
+		course.setAssignments(new java.util.ArrayList<Assignment>());
+		
+		given(courseRepository.findById(TEST_COURSE_ID)).willReturn(Optional.of(course));
+		
+		AssignmentDTO adto = new AssignmentDTO();
+		adto.assignmentName = "NewAssignment";
+		adto.dueDate = "2022-01-01";
+		
+		response = mvc.perform(
+				MockMvcRequestBuilders
+			      .post("/course/40442/add")
```

Comment:
Maybe instead of hardcoding the 40442 you can make String variable for the path that = "/course/" +TEST_COURSE_ID + "/add"

Reply:
yeah I'll do that...I'll have to convert to a string first though since its an integer

Reply:
changes implemented